fullstory-segment-middleware/src/test/java/com/fullstorydev/fullstory_segment_middleware/FullStorySegmentMiddlewareTest.java

```diff
@@ -0,0 +1,442 @@
+
+        // chain.proceed is called with the correct payload
+        verify(mockChain, times(1)).proceed(groupPayload);
+    }
+
+    @Test
+    public void intercept_GroupPayloadChain_DisableGroupTraitsAsUserVars_FSSetUserVarsCalled() {
+        fullStorySegmentMiddleware = new FullStorySegmentMiddleware(mockContext, "SegmentWriteMockKey");
+        fullStorySegmentMiddleware.enableGroupTraitsAsUserVars = false;
+        fullStorySegmentMiddleware.enableFSSessionURLInEvents = false;
+        GroupPayload groupPayload = new GroupPayload.Builder().userId("userId").groupId("groupId").build();
+        when(mockChain.payload()).thenReturn(groupPayload);
+        fullStorySegmentMiddleware.intercept(mockChain);
+
+        // FS.event is called with properties
+        Map<String, String> userVars = new HashMap<>();
+        userVars.put("groupID_str", groupPayload.groupId());
+        verifyStatic(FS.class, VerificationModeFactory.times(1));
+        FS.setUserVars(userVars);
+    }
+
+    @Test
+    public void intercept_GroupPayloadChain_EnableGroupTraitsAsUserVars_ChainProceedCalled() {
+        fullStorySegmentMiddleware = new FullStorySegmentMiddleware(mockContext, "SegmentWriteMockKey");
+        fullStorySegmentMiddleware.enableGroupTraitsAsUserVars = true;
+        fullStorySegmentMiddleware.enableFSSessionURLInEvents = false;
+        Map<String, String> map = new HashMap<>();
+        map.put("industry", "retail");
+        GroupPayload groupPayload = new GroupPayload.Builder()
+                .userId("userId")
+                .groupId("groupId")
+                .traits(map)
+                .build();
+        when(mockChain.payload()).thenReturn(groupPayload);
+        fullStorySegmentMiddleware.intercept(mockChain);
+
+        // chain.proceed is called with the correct payload
+        verify(mockChain, times(1)).proceed(groupPayload);
+    }
+
+    @Test
+    public void intercept_GroupPayloadChain_EnableGroupTraitsAsUserVars_FSSetUserVarsCalled() {
+        fullStorySegmentMiddleware = new FullStorySegmentMiddleware(mockContext, "SegmentWriteMockKey");
+        fullStorySegmentMiddleware.enableGroupTraitsAsUserVars = true;
+        fullStorySegmentMiddleware.enableFSSessionURLInEvents = false;
+        Map<String, String> map = new HashMap<>();
+        map.put("industry", "retail");
+        GroupPayload groupPayload = new GroupPayload.Builder()
+                .userId("userId")
+                .groupId("groupId")
+                .traits(map)
+                .build();
+        when(mockChain.payload()).thenReturn(groupPayload);
+        fullStorySegmentMiddleware.intercept(mockChain);
+
+        // FS.event is called with properties
+        Map<String, String> suffixedMap = new HashMap<>();
+        suffixedMap.put("groupID_str", groupPayload.groupId());
+        suffixedMap.put("industry_str", "retail");
+        verifyStatic(FS.class, VerificationModeFactory.times(1));
+        FS.setUserVars(suffixedMap);
```

Comment:
See comment above. Are you trying to verify that `FS.setUSerVars` is being called in the middleware under certain conditions (like when `fullStorySegmentMiddleware.enableGroupTraitsAsUserVars = true`)?

Reply:
yup, verifying `FS.setUSerVars` is called once with the correct parameters